index out of range [1] with length 1

I recently stumbled on this error prompt.
It might be due to fb forcing me to change password.

I tried modifying debug_login.sh to input my new creds, but I can't make it work. (I don't/can't code)

```
PS C:\Users\jaspe\Documents\GitHub\facebook-delete> go run deleter.go -rateLimit 7000
Waiting 7000 ms before search and delete requests.
panic: runtime error: index out of range [1] with length 1


goroutine 1 [running]:
main.(*fbLogin).StoreProfileID(0xc000064400, {0xc0000fe000, 0xf232ff})
        C:/Users/jaspe/Documents/GitHub/facebook-delete/deleter.go:178 +0xbe
main.(*fbLogin).IsLoggedIn(0xc000064400)
        C:/Users/jaspe/Documents/GitHub/facebook-delete/deleter.go:172 +0x75
main.newFbLogin(0xc000056fb0)
        C:/Users/jaspe/Documents/GitHub/facebook-delete/deleter.go:113 +0x53
main.main()
        C:/Users/jaspe/Documents/GitHub/facebook-delete/deleter.go:522 +0x289
exit status 2
PS C:\Users\jaspe\Documents\GitHub\facebook-delete>
```

--- a/README.md
+++ b/README.md
@@ -2,7 +2,7 @@
 
 This program can be used to clean up a Facebook account without deleting the entire account.
 
-_Warning: Facebook has some measures in place to prevent high-frequency activity such as the one this tool provides. Please take a look at the [Rate limiting](#rate-limiting) section to learn more and try and prevent detection. 
+_Warning: Facebook has some measures in place to prevent high-frequency activity such as the one this tool provides. Please take a look at the [Rate limiting](#rate-limiting) section to learn more and try and prevent detection._ 
 
 _Note: Facebook has a very strange login process. Please open a GitHub issue if the program is not able to login. Here's a [workaround for the login process](https://github.com/marcelja/facebook-delete/wiki/Login-with-browser-cookie) which you can also use if your account has two-factor authentication enabled._
 

--- a/README.md
+++ b/README.md
@@ -2,7 +2,7 @@
 
 This program can be used to clean up a Facebook account without deleting the entire account.
 
-_Warning: Facebook has some measures in place to prevent high-frequency activity such as the one this tool provides. The current rate limit value has currently been set to 30 seconds to avoid detection, but the delay might still not be enough and this could still result in your activity log getting temporarily blocked. Please check out the [Rate-limiting](#rate-limiting) section if you want to increase/decrease this delay._
+_Warning: Facebook has some measures in place to prevent high-frequency activity such as the one this tool provides. The current rate limit value has currently been set to 100 ms to avoid detection, but the delay might not be enough and this could still result in your activity log getting temporarily blocked. Please check out the [Rate-limiting](#rate-limiting) section if you want to increase/decrease this delay._
 
 _Note: Facebook has a very strange login process. Please open a GitHub issue if the program is not able to login. Here's a [workaround for the login process](https://github.com/marcelja/facebook-delete/wiki/Login-with-browser-cookie) which you can also use if your account has two-factor authentication enabled._
 
@@ -46,7 +46,7 @@ Cookies are saved to `$HOME/.go-cookies` if the `$GOCOOKIES` variable is not set
 
 ### Rate-limiting
 
-Facebook will temp-block your activity log if you make too many requests too quickly. Because of this, the current default delay has been set to 30 seconds (30000 ms). Run the command with the `-rateLimit <time in ms>` to introduce a delay before each request. If you're getting hung up on just searches or deletes, you can disable rate-limiting for one or the other with `-limitSearch=0` or `-limitDelete=0`.
+Facebook will temp-block your activity log if you make too many requests too quickly. Run the command with the `-rateLimit <time in ms>` to introduce a custom delay before each request. If you're getting hung up on just searches or deletes, you can disable rate-limiting for one or the other with `-limitSearch=0` or `-limitDelete=0`.
 
 Usage examples:
 

--- a/README.md
+++ b/README.md
@@ -1,6 +1,8 @@
 # Fast Facebook Activity Deletion
 
-This program can be used to clean up a facebook account without deleting the entire account.
+This program can be used to clean up a Facebook account without deleting the entire account.
+
+_Warning: Facebook has some measures in place to prevent high-frequency activity such as the one this tool provides. Please take a look at the [Rate limiting](#rate-limiting) section to learn more and try and prevent detection. 
 
 _Note: Facebook has a very strange login process. Please open a GitHub issue if the program is not able to login. Here's a [workaround for the login process](https://github.com/marcelja/facebook-delete/wiki/Login-with-browser-cookie) which you can also use if your account has two-factor authentication enabled._
 

--- a/README.md
+++ b/README.md
@@ -2,7 +2,7 @@
 
 This program can be used to clean up a Facebook account without deleting the entire account.
 
-_Warning: Facebook has some measures in place to prevent high-frequency activity such as the one this tool provides. The current rate limit has currently been set to 30 seconds to avoid detection, but it might not even be slow enough and still result in your activity log getting temporarily blocked. Please check out the [Rate-limiting](#rate-limiting) section if you want to increase/decrease this limit._ 
+_Warning: Facebook has some measures in place to prevent high-frequency activity such as the one this tool provides. The current rate limit value has currently been set to 30 seconds to avoid detection, but the delay might still not be enough and this could still result in your activity log getting temporarily blocked. Please check out the [Rate-limiting](#rate-limiting) section if you want to increase/decrease this delay._ 
 
 _Note: Facebook has a very strange login process. Please open a GitHub issue if the program is not able to login. Here's a [workaround for the login process](https://github.com/marcelja/facebook-delete/wiki/Login-with-browser-cookie) which you can also use if your account has two-factor authentication enabled._
 
@@ -46,7 +46,7 @@ Cookies are saved to `$HOME/.go-cookies` if the `$GOCOOKIES` variable is not set
 
 ### Rate-limiting
 
-Facebook will temp-block you if you make too many requests too quickly. Run the command with the `-rateLimit <time in ms>` to introduce a delay before each request. If you're getting hung up on just searches or deletes, you can disable rate-limiting for one or the other with `-limitSearch=0` or `-limitDelete=0`.
+Facebook will temp-block your activity log if you make too many requests too quickly. Because of this, the current default delay has been set to 30 seconds (30000 ms). Run the command with the `-rateLimit <time in ms>` to introduce a delay before each request. If you're getting hung up on just searches or deletes, you can disable rate-limiting for one or the other with `-limitSearch=0` or `-limitDelete=0`.
 
 E.g. one of:
 

--- a/README.md
+++ b/README.md
@@ -2,11 +2,11 @@
 
 This program can be used to clean up a Facebook account without deleting the entire account.
 
-_Warning: Facebook has some measures in place to prevent high-frequency activity such as the one this tool provides. The current rate limit value has currently been set to 30 seconds to avoid detection, but the delay might still not be enough and this could still result in your activity log getting temporarily blocked. Please check out the [Rate-limiting](#rate-limiting) section if you want to increase/decrease this delay._ 
+_Warning: Facebook has some measures in place to prevent high-frequency activity such as the one this tool provides. The current rate limit value has currently been set to 30 seconds to avoid detection, but the delay might still not be enough and this could still result in your activity log getting temporarily blocked. Please check out the [Rate-limiting](#rate-limiting) section if you want to increase/decrease this delay._
 
 _Note: Facebook has a very strange login process. Please open a GitHub issue if the program is not able to login. Here's a [workaround for the login process](https://github.com/marcelja/facebook-delete/wiki/Login-with-browser-cookie) which you can also use if your account has two-factor authentication enabled._
 
-![](demo.gif)
+![Demo](demo.gif)
 
 ## Run program
 
@@ -18,9 +18,9 @@ Download the binary file for your platform of the [latest release](https://githu
 
 Linux/macOS: Make the binary executable and run it. Example:
 
-```
-$ chmod +x deleter-linux
-$ ./deleter-linux
+```bash
+chmod +x deleter-linux
+./deleter-linux
 ```
 
 Windows: Run the .exe file and select "More information" and "Run anyway".
@@ -29,18 +29,18 @@ Windows: Run the .exe file and select "More information" and "Run anyway".
 
 A recent [Go](https://golang.org/) version needs to be installed. This can be done via your package manager or the golang website.
 
-__Clone this repository__
+#### Clone this repository
 
-```
-$ git clone https://github.com/marcelja/facebook-delete.git
-$ cd facebook-delete
-$ go install
-$ go run deleter.go
+```bash
+git clone https://github.com/marcelja/facebook-delete.git
+cd facebook-delete
+go install
+go run deleter.go
 ```
 
 ## Cookies
 
-Cookies are saved to `$HOME/.go-cookies` if the `$GOCOOKIES` variable is not set (see https://github.com/juju/persistent-cookiejar).
+Cookies are saved to `$HOME/.go-cookies` if the `$GOCOOKIES` variable is not set (see <https://github.com/juju/persistent-cookiejar>).
 
 ## Options
 
@@ -48,12 +48,12 @@ Cookies are saved to `$HOME/.go-cookies` if the `$GOCOOKIES` variable is not set
 
 Facebook will temp-block your activity log if you make too many requests too quickly. Because of this, the current default delay has been set to 30 seconds (30000 ms). Run the command with the `-rateLimit <time in ms>` to introduce a delay before each request. If you're getting hung up on just searches or deletes, you can disable rate-limiting for one or the other with `-limitSearch=0` or `-limitDelete=0`.
 
-E.g. one of:
+Usage examples:
 
-```
-$ ./deleter-linux -rateLimit 500
-$ ./deleter-linux -rateLimit 500 -limitSearch=0
-$ ./deleter-linux -rateLimit 500 -limitDelete=0
+```bash
+./deleter-linux -rateLimit 500 #Add a 500 ms delay
+./deleter-linux -rateLimit 500 -limitSearch=0 #Add a 500 ms delay, but not to search action
+./deleter-linux -rateLimit 500 -limitDelete=0 # Add a 500 ms delay, but not to deletion action
 ```
 
 ### Selection via flags

--- a/README.md
+++ b/README.md
@@ -2,7 +2,7 @@
 
 This program can be used to clean up a Facebook account without deleting the entire account.
 
-_Warning: Facebook has some measures in place to prevent high-frequency activity such as the one this tool provides. The current rate limit has been currently set to 30 seconds, which might not even be slow enough. Please check out the [Rate-limiting](#rate-limiting) section if you want to increase/decrease this limit._ 
+_Warning: Facebook has some measures in place to prevent high-frequency activity such as the one this tool provides. The current rate limit has currently been set to 30 seconds to avoid detection, but it might not even be slow enough and still result in your activity log getting temporarily blocked. Please check out the [Rate-limiting](#rate-limiting) section if you want to increase/decrease this limit._ 
 
 _Note: Facebook has a very strange login process. Please open a GitHub issue if the program is not able to login. Here's a [workaround for the login process](https://github.com/marcelja/facebook-delete/wiki/Login-with-browser-cookie) which you can also use if your account has two-factor authentication enabled._
 

--- a/README.md
+++ b/README.md
@@ -54,3 +54,12 @@ $ ./deleter-linux -rateLimit 500 -limitSearch=0
 $ ./deleter-linux -rateLimit 500 -limitDelete=0
 ```
 
+### Selection via flags
+
+| Flag              | Type     | Description                                                             | Example                                                 |
+|-------------------|----------|-------------------------------------------------------------------------|---------------------------------------------------------|
+| -customYears      | `string` | Comma-separated years (YYYY) to select. Use `all` for a full selection. | `-customYears="2006,2009,2020"` or `-customYears="all"` |
+| -customMonths     | `string` | Comma-separated months to select.                                       | `-customMonths="1,2,12` or `-customMonths="01,02,12"`   |
+| -selectAllContent | `bool`   | If set to `true` it will select all content without asking questions.   | `-selectAllContent=true`                                |
+
+Note: if invalid arguments are passed to the flags, you will be warned and asked to do a manual selection.

--- a/README.md
+++ b/README.md
@@ -2,7 +2,7 @@
 
 This program can be used to clean up a Facebook account without deleting the entire account.
 
-_Warning: Facebook has some measures in place to prevent high-frequency activity such as the one this tool provides. Please take a look at the [Rate limiting](#rate-limiting) section to learn more and try and prevent detection._ 
+_Warning: Facebook has some measures in place to prevent high-frequency activity such as the one this tool provides. The current rate limit has been currently set to 30 seconds, which might not even be slow enough. Please check out the [Rate-limiting](#rate-limiting) section if you want to increase/decrease this limit._ 
 
 _Note: Facebook has a very strange login process. Please open a GitHub issue if the program is not able to login. Here's a [workaround for the login process](https://github.com/marcelja/facebook-delete/wiki/Login-with-browser-cookie) which you can also use if your account has two-factor authentication enabled._
 
@@ -58,10 +58,12 @@ $ ./deleter-linux -rateLimit 500 -limitDelete=0
 
 ### Selection via flags
 
+If you want to skip the manual terminal selection, feel free to use the following flags:
+
 | Flag              | Type     | Description                                                            | Example                                                 |
 |-------------------|----------|------------------------------------------------------------------------|---------------------------------------------------------|
 | -customYears      | `string` | Comma-separated years (YYYY) to select. Use `all` to select all years. | `-customYears="2006,2009,2020"` or `-customYears="all"` |
 | -customMonths     | `string` | Comma-separated months to select. Use `all` to select all months.      | `-customMonths="1,2,12` or `-customMonths="01,02,12"`   |
 | -selectAllContent | `bool`   | If set to `true` it will select all content without asking questions.  | `-selectAllContent=true`                                |                              |
 
-Note: if invalid arguments are passed to the flags, you will be warned and asked to do a manual selection.
+_Note: if invalid arguments are passed to the flags, a warning will appear and you will be asked to perform a manual selection._

--- a/README.md
+++ b/README.md
@@ -58,10 +58,10 @@ $ ./deleter-linux -rateLimit 500 -limitDelete=0
 
 ### Selection via flags
 
-| Flag              | Type     | Description                                                             | Example                                                 |
-|-------------------|----------|-------------------------------------------------------------------------|---------------------------------------------------------|
-| -customYears      | `string` | Comma-separated years (YYYY) to select. Use `all` for a full selection. | `-customYears="2006,2009,2020"` or `-customYears="all"` |
-| -customMonths     | `string` | Comma-separated months to select.                                       | `-customMonths="1,2,12` or `-customMonths="01,02,12"`   |
-| -selectAllContent | `bool`   | If set to `true` it will select all content without asking questions.   | `-selectAllContent=true`                                |
+| Flag              | Type     | Description                                                            | Example                                                 |
+|-------------------|----------|------------------------------------------------------------------------|---------------------------------------------------------|
+| -customYears      | `string` | Comma-separated years (YYYY) to select. Use `all` to select all years. | `-customYears="2006,2009,2020"` or `-customYears="all"` |
+| -customMonths     | `string` | Comma-separated months to select. Use `all` to select all months.      | `-customMonths="1,2,12` or `-customMonths="01,02,12"`   |
+| -selectAllContent | `bool`   | If set to `true` it will select all content without asking questions.  | `-selectAllContent=true`                                |                              |
 
 Note: if invalid arguments are passed to the flags, you will be warned and asked to do a manual selection.

--- a/deleter.go
+++ b/deleter.go
@@ -526,7 +526,7 @@ func validateMonthsFlag(flagContent string) bool {
 }
 
 func main() {
-	flag.IntVar(&rateLimit, "rateLimit", 30000, "Wait this many milliseconds between requests.")
+	flag.IntVar(&rateLimit, "rateLimit", 100, "Wait this many milliseconds between requests.")
 	flag.BoolVar(&limitSearch, "limitSearch", true, "Rate-limit searching for things to delete.")
 	flag.BoolVar(&limitDelete, "limitDelete", true, "Rate-limit deleting things.")
 	flag.StringVar(&customYears, "customYears", "", "Comma-separated years (YYYY) to select.")

--- a/deleter.go
+++ b/deleter.go
@@ -510,10 +510,10 @@ func (del *deleter) DeleteElement(elem *deleteElement) {
 }
 
 func main() {
-	flag.IntVar(&rateLimit, "rateLimit", 0, "Wait this many milliseconds between requests.")
+	flag.IntVar(&rateLimit, "rateLimit", 30000, "Wait this many milliseconds between requests.")
 	flag.BoolVar(&limitSearch, "limitSearch", true, "Rate-limit searching for things to delete.")
 	flag.BoolVar(&limitDelete, "limitDelete", true, "Rate-limit deleting things.")
-	flag.StringVar(&customYears, "customYears", "", "Comma-separated years (YYYY) to set.")
+	flag.StringVar(&customYears, "customYears", "", "Comma-separated years (YYYY) to select.")
 	flag.StringVar(&customMonths, "customMonths", "", "Comma-separated months (MM) to set.")
 	flag.BoolVar(&selectAllContent, "selectAllContent", false, "Don't ask content type, but select all.")
 	flag.BoolVar(&excludeInstagram, "excludeInstagram", false, "Exclude Instagram from content.")

--- a/deleter.go
+++ b/deleter.go
@@ -544,8 +544,8 @@ func main() {
 	customMonths = strings.ReplaceAll(customMonths, " ", "")
 	re2, _ := regexp.Compile("^((0?[1-9]|1[012])([,](0?[1-9]|1[012]))*)?$")
 	if !re2.MatchString(customMonths) {
-		if customYears == "all" {
-			customMonths = "Jan,Feb,Mar,May,Apr,Jun,Jul,Aug,Sep,Oct,Nov,Dec"
+		if customMonths == "all" {
+			customMonths = "01,02,03,04,05,06,07,08,09,10,11,12"
 		} else {
 			fmt.Println("Invalid months passed through customMonths flag.")
 			customMonths = ""

--- a/deleter.go
+++ b/deleter.go
@@ -17,7 +17,6 @@ import (
 	"github.com/AlecAivazis/survey/v2"
 	"github.com/cheggaaa/pb/v3"
 	"github.com/juju/persistent-cookiejar"
-	//"net/http/cookiejar"
 )
 
 const numRoutines int = 5
@@ -187,7 +186,6 @@ func (fbl *fbLogin) StoreProfileID(output string) {
 		panic(fmt.Sprintf("Unable to parse profile information from cookies. Try manually deleting the cookie file stored in %s", cookiejar.DefaultCookieFile()))
 	}
 	result := strings.Split(output, ";profile_id=")[1]
-	fmt.Println("Result:", result)
 	result = strings.Split(result, "&amp;")[0]
 	fbl.profileID = result
 }
@@ -527,70 +525,6 @@ func validateMonthsFlag(flagContent string) bool {
 	return re.MatchString(months) || flagContent == "all" || flagContent == ""
 }
 
-func processYears(flagContent string) []string {
-	years := []string{}
-	if flagContent == "" {
-		years = createMultiSelect("years", yearOptions)
-	} else {
-		years = strings.Split(flagContent, ",")
-		fmt.Printf("? Which years: %s \n", strings.Join(years, ", "))
-	}
-	return years
-}
-
-func processMonths(flagContent string, actRead activityReader) {
-	months := []string{}
-	if flagContent == "" {
-		months = createMultiSelect("months", monthStrings)
-		actRead.selectedMonths = months
-	} else {
-		months = strings.Split(flagContent, ",")
-		monthLabels := make(map[string]string)
-		monthLabels["1"] = "Jan"
-		monthLabels["01"] = "Jan"
-		monthLabels["2"] = "Feb"
-		monthLabels["02"] = "Feb"
-		monthLabels["3"] = "Mar"
-		monthLabels["03"] = "Mar"
-		monthLabels["4"] = "Apr"
-		monthLabels["04"] = "Apr"
-		monthLabels["5"] = "May"
-		monthLabels["05"] = "May"
-		monthLabels["6"] = "Jun"
-		monthLabels["06"] = "Jun"
-		monthLabels["7"] = "Jul"
-		monthLabels["07"] = "Jul"
-		monthLabels["8"] = "Aug"
-		monthLabels["08"] = "Aug"
-		monthLabels["9"] = "Sep"
-		monthLabels["09"] = "Sep"
-		monthLabels["10"] = "Oct"
-		monthLabels["11"] = "Nov"
-		monthLabels["12"] = "Dec"
-
-		newMonths := []string{}
-		for _, month := range months {
-			if month, found := monthLabels[month]; found {
-				newMonths = append(newMonths, month)
-			}
-		}
-		fmt.Printf("? Which months: %s \n", strings.Join(newMonths, ", "))
-		actRead.selectedMonths = newMonths
-	}
-}
-
-func processCategories(reader activityReader) []string {
-	categories := []string{}
-	if selectAllContent {
-		categories = categorySlice()
-		fmt.Printf("Selected content: %s \n", strings.Join(categories, ", "))
-	} else {
-		categories = createMultiSelect("categories", categorySlice())
-	}
-
-	return categories
-}
-
 func main() {
 	flag.IntVar(&rateLimit, "rateLimit", 30000, "Wait this many milliseconds between requests.")
 	flag.BoolVar(&limitSearch, "limitSearch", true, "Rate-limit searching for things to delete.")
@@ -637,9 +571,62 @@ func main() {
 		customMonths = ""
 	}
 
-	years := processYears(customYears)
-	processMonths(customMonths, actRead)
-	categories := processCategories(actRead)
+	// Data processing
+
+	years := []string{}
+	if customYears == "" {
+		years = createMultiSelect("years", yearOptions)
+	} else {
+		years = strings.Split(customYears, ",")
+		fmt.Printf("? Which years: %s \n", strings.Join(years, ", "))
+	}
+
+	months := []string{}
+	if customMonths == "" {
+		months = createMultiSelect("months", monthStrings)
+		actRead.selectedMonths = months
+	} else {
+		months = strings.Split(customMonths, ",")
+		monthLabels := make(map[string]string)
+		monthLabels["1"] = "Jan"
+		monthLabels["01"] = "Jan"
+		monthLabels["2"] = "Feb"
+		monthLabels["02"] = "Feb"
+		monthLabels["3"] = "Mar"
+		monthLabels["03"] = "Mar"
+		monthLabels["4"] = "Apr"
+		monthLabels["04"] = "Apr"
+		monthLabels["5"] = "May"
+		monthLabels["05"] = "May"
+		monthLabels["6"] = "Jun"
+		monthLabels["06"] = "Jun"
+		monthLabels["7"] = "Jul"
+		monthLabels["07"] = "Jul"
+		monthLabels["8"] = "Aug"
+		monthLabels["08"] = "Aug"
+		monthLabels["9"] = "Sep"
+		monthLabels["09"] = "Sep"
+		monthLabels["10"] = "Oct"
+		monthLabels["11"] = "Nov"
+		monthLabels["12"] = "Dec"
+
+		newMonths := []string{}
+		for _, month := range months {
+			if month, found := monthLabels[month]; found {
+				newMonths = append(newMonths, month)
+			}
+		}
+		fmt.Printf("? Which months: %s \n", strings.Join(newMonths, ", "))
+		actRead.selectedMonths = newMonths
+	}
+
+	categories := []string{}
+	if selectAllContent {
+		categories = categorySlice()
+		fmt.Printf("Selected content: %s \n", strings.Join(categories, ", "))
+	} else {
+		categories = createMultiSelect("categories", categorySlice())
+	}
 
 	del := deleter{&actRead, req}
 	del.Delete(years, categories)

--- a/deleter.go
+++ b/deleter.go
@@ -53,7 +53,6 @@ var limitDelete bool
 var customYears string
 var customMonths string
 var selectAllContent bool
-var excludeInstagram bool
 
 type requester struct {
 	client *http.Client
@@ -516,7 +515,6 @@ func main() {
 	flag.StringVar(&customYears, "customYears", "", "Comma-separated years (YYYY) to select.")
 	flag.StringVar(&customMonths, "customMonths", "", "Comma-separated months (MM) to set.")
 	flag.BoolVar(&selectAllContent, "selectAllContent", false, "Don't ask content type, but select all.")
-	flag.BoolVar(&excludeInstagram, "excludeInstagram", false, "Exclude Instagram from content.")
 	flag.Parse()
 	if rateLimit > 0 {
 		if limitSearch && limitDelete {
@@ -526,10 +524,6 @@ func main() {
 		} else if limitDelete {
 			fmt.Printf("Waiting %d ms before delete requests.\n", rateLimit)
 		}
-	}
-	if excludeInstagram == true {
-		fmt.Println("Instagram content will be excluded.")
-		delete(categoriesMap, "Instagram Photos and Videos")
 	}
 
 	// Validate custom years flag
@@ -563,6 +557,7 @@ func main() {
 	fbl := newFbLogin(req)
 	actRead := activityReader{req, fbl, make([]deleteElement, 0), make([]string, 0)}
 
+	// Process Years
 	if customYears == "" {
 		years = createMultiSelect("years", yearOptions)
 	} else {
@@ -570,6 +565,7 @@ func main() {
 		fmt.Printf("Selected years: %s \n", strings.Join(years, ", "))
 	}
 
+	// Process Months
 	if customMonths == "" {
 		months = createMultiSelect("months", monthStrings)
 		actRead.selectedMonths = months
@@ -609,6 +605,7 @@ func main() {
 		actRead.selectedMonths = newMonths
 	}
 
+	// Process Categories
 	categories := []string{}
 	if selectAllContent {
 		categories = categorySlice()
@@ -616,6 +613,7 @@ func main() {
 		categories = createMultiSelect("categories", categorySlice())
 	}
 
+	// Delete
 	del := deleter{&actRead, req}
 	del.Delete(years, categories)
 }

--- a/deleter.go
+++ b/deleter.go
@@ -4,17 +4,19 @@ package main
 import (
 	"flag"
 	"fmt"
-	"github.com/AlecAivazis/survey/v2"
-	"github.com/cheggaaa/pb/v3"
-	"github.com/juju/persistent-cookiejar"
 	"io/ioutil"
 	"net/http"
+	"net/http/cookiejar"
 	"net/url"
+	"regexp"
 	"sort"
 	"strconv"
 	"strings"
 	"sync"
 	"time"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/cheggaaa/pb/v3"
 )
 
 const numRoutines int = 5
@@ -48,6 +50,10 @@ var tokensInURLs = [...]string{"/removecontent", "/delete", "/report", "/events/
 var rateLimit int
 var limitSearch bool
 var limitDelete bool
+var customYears string
+var customMonths string
+var selectAllContent bool
+var excludeInstagram bool
 
 type requester struct {
 	client *http.Client
@@ -113,7 +119,7 @@ func newFbLogin(req *requester) *fbLogin {
 	if !fbl.IsLoggedIn() {
 		fbl.EnterInformation()
 		fbl.Login()
-		req.jar.Save()
+		//req.jar.Save()
 		if !fbl.IsLoggedIn() {
 			panic("Failed to login. Please open https://mbasic.facebook.com in a browser and login there. Facebook might requires an additional verification. Afterwards, you can try again.")
 		}
@@ -507,6 +513,10 @@ func main() {
 	flag.IntVar(&rateLimit, "rateLimit", 0, "Wait this many milliseconds between requests.")
 	flag.BoolVar(&limitSearch, "limitSearch", true, "Rate-limit searching for things to delete.")
 	flag.BoolVar(&limitDelete, "limitDelete", true, "Rate-limit deleting things.")
+	flag.StringVar(&customYears, "customYears", "", "Comma-separated years (YYYY) to set.")
+	flag.StringVar(&customMonths, "customMonths", "", "Comma-separated months (MM) to set.")
+	flag.BoolVar(&selectAllContent, "selectAllContent", false, "Don't ask content type, but select all.")
+	flag.BoolVar(&excludeInstagram, "excludeInstagram", false, "Exclude Instagram from content.")
 	flag.Parse()
 	if rateLimit > 0 {
 		if limitSearch && limitDelete {
@@ -517,15 +527,94 @@ func main() {
 			fmt.Printf("Waiting %d ms before delete requests.\n", rateLimit)
 		}
 	}
+	if excludeInstagram == true {
+		fmt.Println("Instagram content will be excluded.")
+		delete(categoriesMap, "Instagram Photos and Videos")
+	}
 
+	// Validate custom years flag
+	years := []string{}
+	customYears = strings.ReplaceAll(customYears, " ", "")
+	re1, _ := regexp.Compile("^(([\\d]{4})([,][\\d]{4})*)?$")
+	if !re1.MatchString(customYears) {
+		if customYears == "all" {
+			customYears = "2022,2021,2020,2019,2018,2017,2016,2015,2014,2013,2012,2011,2010,2009,2008,2007,2006"
+		} else {
+			fmt.Println("Invalid years passed through customYears flag.")
+			customYears = ""
+		}
+	}
+
+	// Validate custom months flag
+	months := []string{}
+	customMonths = strings.ReplaceAll(customMonths, " ", "")
+	re2, _ := regexp.Compile("^((0?[1-9]|1[012])([,](0?[1-9]|1[012]))*)?$")
+	if !re2.MatchString(customMonths) {
+		if customYears == "all" {
+			customMonths = "Jan,Feb,Mar,May,Apr,Jun,Jul,Aug,Sep,Oct,Nov,Dec"
+		} else {
+			fmt.Println("Invalid months passed through customMonths flag.")
+			customMonths = ""
+		}
+	}
+
+	// Login
 	req := newRequester()
 	fbl := newFbLogin(req)
 	actRead := activityReader{req, fbl, make([]deleteElement, 0), make([]string, 0)}
 
-	years := createMultiSelect("years", yearOptions)
-	months := createMultiSelect("months", monthStrings)
-	actRead.selectedMonths = months
-	categories := createMultiSelect("categories", categorySlice())
+	if customYears == "" {
+		years = createMultiSelect("years", yearOptions)
+	} else {
+		years = strings.Split(customYears, ",")
+		fmt.Printf("Selected years: %s \n", strings.Join(years, ", "))
+	}
+
+	if customMonths == "" {
+		months = createMultiSelect("months", monthStrings)
+		actRead.selectedMonths = months
+	} else {
+		months = strings.Split(customMonths, ",")
+		monthLabels := make(map[string]string)
+		monthLabels["1"] = "Jan"
+		monthLabels["01"] = "Jan"
+		monthLabels["2"] = "Feb"
+		monthLabels["02"] = "Feb"
+		monthLabels["3"] = "Mar"
+		monthLabels["03"] = "Mar"
+		monthLabels["4"] = "Apr"
+		monthLabels["04"] = "Apr"
+		monthLabels["5"] = "May"
+		monthLabels["05"] = "May"
+		monthLabels["6"] = "Jun"
+		monthLabels["06"] = "Jun"
+		monthLabels["7"] = "Jul"
+		monthLabels["07"] = "Jul"
+		monthLabels["8"] = "Aug"
+		monthLabels["08"] = "Aug"
+		monthLabels["9"] = "Sep"
+		monthLabels["09"] = "Sep"
+		monthLabels["10"] = "Oct"
+		monthLabels["11"] = "Nov"
+		monthLabels["12"] = "Dec"
+
+		newMonths := []string{}
+		for _, month := range months {
+			if month, found := monthLabels[month]; found {
+				fmt.Printf("Found month: %s \n", month)
+				newMonths = append(newMonths, month)
+			}
+		}
+		fmt.Printf("Selected months: %s \n", strings.Join(newMonths, ", "))
+		actRead.selectedMonths = newMonths
+	}
+
+	categories := []string{}
+	if selectAllContent {
+		categories = categorySlice()
+	} else {
+		categories = createMultiSelect("categories", categorySlice())
+	}
 
 	del := deleter{&actRead, req}
 	del.Delete(years, categories)

--- a/deleter.go
+++ b/deleter.go
@@ -16,8 +16,8 @@ import (
 
 	"github.com/AlecAivazis/survey/v2"
 	"github.com/cheggaaa/pb/v3"
-	//"github.com/juju/persistent-cookiejar"
-	"net/http/cookiejar"
+	"github.com/juju/persistent-cookiejar"
+	//"net/http/cookiejar"
 )
 
 const numRoutines int = 5
@@ -100,7 +100,7 @@ func retrieveRequestString(resp *http.Response, err error) string {
 	}
 	strBody := string(body)
 	if strings.Contains(strBody, "You can try again later") || strings.Contains(strBody, "temporarily blocked") {
-		panic("ratelimited, please open https://mbasic.facebook.com and navigate to your activity log to see more information.")
+		panic("Ratelimited, please open https://mbasic.facebook.com and navigate to your activity log to see more information.")
 	}
 	return strBody
 }
@@ -119,7 +119,7 @@ func newFbLogin(req *requester) *fbLogin {
 	if !fbl.IsLoggedIn() {
 		fbl.EnterInformation()
 		fbl.Login()
-		//req.jar.Save()
+		req.jar.Save()
 		if !fbl.IsLoggedIn() {
 			panic("Failed to login. Please open https://mbasic.facebook.com in a browser and login there. Facebook might requires an additional verification. Afterwards, you can try again.")
 		}
@@ -181,7 +181,13 @@ func (fbl *fbLogin) IsLoggedIn() bool {
 }
 
 func (fbl *fbLogin) StoreProfileID(output string) {
+	slice := strings.Split(output, ";profile_id=")
+	if len(slice) == 1 {
+		fbl.requester.jar.RemoveAll()
+		panic(fmt.Sprintf("Unable to parse profile information from cookies. Try manually deleting the cookie file stored in %s", cookiejar.DefaultCookieFile()))
+	}
 	result := strings.Split(output, ";profile_id=")[1]
+	fmt.Println("Result:", result)
 	result = strings.Split(result, "&amp;")[0]
 	fbl.profileID = result
 }

--- a/deleter.go
+++ b/deleter.go
@@ -6,7 +6,6 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net/http"
-	"net/http/cookiejar"
 	"net/url"
 	"regexp"
 	"sort"
@@ -17,6 +16,8 @@ import (
 
 	"github.com/AlecAivazis/survey/v2"
 	"github.com/cheggaaa/pb/v3"
+	//"github.com/juju/persistent-cookiejar"
+	"net/http/cookiejar"
 )
 
 const numRoutines int = 5
@@ -508,69 +509,36 @@ func (del *deleter) DeleteElement(elem *deleteElement) {
 	}
 }
 
-func main() {
-	flag.IntVar(&rateLimit, "rateLimit", 30000, "Wait this many milliseconds between requests.")
-	flag.BoolVar(&limitSearch, "limitSearch", true, "Rate-limit searching for things to delete.")
-	flag.BoolVar(&limitDelete, "limitDelete", true, "Rate-limit deleting things.")
-	flag.StringVar(&customYears, "customYears", "", "Comma-separated years (YYYY) to select.")
-	flag.StringVar(&customMonths, "customMonths", "", "Comma-separated months (MM) to set.")
-	flag.BoolVar(&selectAllContent, "selectAllContent", false, "Don't ask content type, but select all.")
-	flag.Parse()
-	if rateLimit > 0 {
-		if limitSearch && limitDelete {
-			fmt.Printf("Waiting %d ms before search and delete requests.\n", rateLimit)
-		} else if limitSearch {
-			fmt.Printf("Waiting %d ms before search requests.\n", rateLimit)
-		} else if limitDelete {
-			fmt.Printf("Waiting %d ms before delete requests.\n", rateLimit)
-		}
-	}
+func validateYearsFlag(flagContent string) bool {
+	years := strings.ReplaceAll(flagContent, " ", "")
+	re, _ := regexp.Compile("^(([\\d]{4})([,][\\d]{4})*)?$")
+	return re.MatchString(years) || flagContent == "all" || flagContent == ""
+}
 
-	// Validate custom years flag
+func validateMonthsFlag(flagContent string) bool {
+	months := strings.ReplaceAll(flagContent, " ", "")
+	re, _ := regexp.Compile("^((0?[1-9]|1[012])([,](0?[1-9]|1[012]))*)?$")
+	return re.MatchString(months) || flagContent == "all" || flagContent == ""
+}
+
+func processYears(flagContent string) []string {
 	years := []string{}
-	customYears = strings.ReplaceAll(customYears, " ", "")
-	re1, _ := regexp.Compile("^(([\\d]{4})([,][\\d]{4})*)?$")
-	if !re1.MatchString(customYears) {
-		if customYears == "all" {
-			customYears = "2022,2021,2020,2019,2018,2017,2016,2015,2014,2013,2012,2011,2010,2009,2008,2007,2006"
-		} else {
-			fmt.Println("Invalid years passed through customYears flag.")
-			customYears = ""
-		}
-	}
-
-	// Validate custom months flag
-	months := []string{}
-	customMonths = strings.ReplaceAll(customMonths, " ", "")
-	re2, _ := regexp.Compile("^((0?[1-9]|1[012])([,](0?[1-9]|1[012]))*)?$")
-	if !re2.MatchString(customMonths) {
-		if customMonths == "all" {
-			customMonths = "01,02,03,04,05,06,07,08,09,10,11,12"
-		} else {
-			fmt.Println("Invalid months passed through customMonths flag.")
-			customMonths = ""
-		}
-	}
-
-	// Login
-	req := newRequester()
-	fbl := newFbLogin(req)
-	actRead := activityReader{req, fbl, make([]deleteElement, 0), make([]string, 0)}
-
-	// Process Years
-	if customYears == "" {
+	if flagContent == "" {
 		years = createMultiSelect("years", yearOptions)
 	} else {
-		years = strings.Split(customYears, ",")
-		fmt.Printf("Selected years: %s \n", strings.Join(years, ", "))
+		years = strings.Split(flagContent, ",")
+		fmt.Printf("? Which years: %s \n", strings.Join(years, ", "))
 	}
+	return years
+}
 
-	// Process Months
-	if customMonths == "" {
+func processMonths(flagContent string, actRead activityReader) {
+	months := []string{}
+	if flagContent == "" {
 		months = createMultiSelect("months", monthStrings)
 		actRead.selectedMonths = months
 	} else {
-		months = strings.Split(customMonths, ",")
+		months = strings.Split(flagContent, ",")
 		monthLabels := make(map[string]string)
 		monthLabels["1"] = "Jan"
 		monthLabels["01"] = "Jan"
@@ -597,23 +565,76 @@ func main() {
 		newMonths := []string{}
 		for _, month := range months {
 			if month, found := monthLabels[month]; found {
-				fmt.Printf("Found month: %s \n", month)
 				newMonths = append(newMonths, month)
 			}
 		}
-		fmt.Printf("Selected months: %s \n", strings.Join(newMonths, ", "))
+		fmt.Printf("? Which months: %s \n", strings.Join(newMonths, ", "))
 		actRead.selectedMonths = newMonths
 	}
+}
 
-	// Process Categories
+func processCategories(reader activityReader) []string {
 	categories := []string{}
 	if selectAllContent {
 		categories = categorySlice()
+		fmt.Printf("Selected content: %s \n", strings.Join(categories, ", "))
 	} else {
 		categories = createMultiSelect("categories", categorySlice())
 	}
 
-	// Delete
+	return categories
+}
+
+func main() {
+	flag.IntVar(&rateLimit, "rateLimit", 30000, "Wait this many milliseconds between requests.")
+	flag.BoolVar(&limitSearch, "limitSearch", true, "Rate-limit searching for things to delete.")
+	flag.BoolVar(&limitDelete, "limitDelete", true, "Rate-limit deleting things.")
+	flag.StringVar(&customYears, "customYears", "", "Comma-separated years (YYYY) to select.")
+	flag.StringVar(&customMonths, "customMonths", "", "Comma-separated months (MM) to set.")
+	flag.BoolVar(&selectAllContent, "selectAllContent", false, "Don't ask content type, but select all.")
+	flag.Parse()
+	if rateLimit > 0 {
+		if limitSearch && limitDelete {
+			fmt.Printf("Waiting %d ms before search and delete requests.\n", rateLimit)
+		} else if limitSearch {
+			fmt.Printf("Waiting %d ms before search requests.\n", rateLimit)
+		} else if limitDelete {
+			fmt.Printf("Waiting %d ms before delete requests.\n", rateLimit)
+		}
+	}
+
+	// Login
+	req := newRequester()
+	fbl := newFbLogin(req)
+	actRead := activityReader{req, fbl, make([]deleteElement, 0), make([]string, 0)}
+
+	// Flags validation
+
+	customYearsFlagValid := validateYearsFlag(customYears)
+	customMonthsFlagValid := validateMonthsFlag(customMonths)
+
+	if customYearsFlagValid {
+		if customYears == "all" {
+			customYears = "2022,2021,2020,2019,2018,2017,2016,2015,2014,2013,2012,2011,2010,2009,2008,2007,2006"
+		}
+	} else {
+		fmt.Println("Invalid years passed through customYears flag.")
+		customYears = ""
+	}
+
+	if customMonthsFlagValid {
+		if customMonths == "all" {
+			customMonths = "01,02,03,04,05,06,07,08,09,10,11,12"
+		}
+	} else {
+		fmt.Println("Invalid months passed through customMonths flag.")
+		customMonths = ""
+	}
+
+	years := processYears(customYears)
+	processMonths(customMonths, actRead)
+	categories := processCategories(actRead)
+
 	del := deleter{&actRead, req}
 	del.Delete(years, categories)
 }

